### ISSUE TITLE
Handling a forum URL intent from another app 'sticks'

### ISSUE THREAD

**Issue by baka-kaba** [Awful/Awful.apk#494]
* get a forum link in another app that doesn't handle URLs internally (e.g. email it to yourself)
* let Awful handle the intent and open the link
* the nav drawer Bookmarks link opens that link instead
* same for the Bookmarks widget
* this happens until the app is closed

**Comment by baka-kaba** [comment]
Ok this is happening in ForumsIndexActivity. When you select the Bookmarks menu option, it creates a new Intent with just a Forum ID and a page number, set to reuse the current activity. So it keeps the same state and that starts to cause issues when the Intent is handled.
Then:
* ``#parseNewIntent`` tries to get the data URI, which would set the activity's ``url`` field - that's currently set to the external link you opened in the app before. This new Bookmarks Intent doesn't have a URI, so the old ``url`` value sticks
* then ``#parseNewIntent`` works out which fragment to show, which ends up being the thread view (not the forum view where the bookmarks are) because it's checking if ``url`` points to a thread, which the old value does
* finally ``#onNewIntent`` checks if ``url`` is a thread (it is) and opens that thread in the thread fragment - but only if the fragment is already open (so it doesn't have any effect when the app is just opening)

So basically the problem here is that ``url`` value sticks around and overrides the behaviour of any subsequent Intent that doesn't replace it

**Comment by baka-kaba** [comment]
Fixed by #583
